packages/e2e-tests/fixtures/blocks/core__pullquote__deprecated-3.serialized.html

```diff
@@ -1,3 +1,3 @@
-<!-- wp:pullquote {"customMainColor":"#cd2653","textColor":"secondary"} -->
-<figure class="wp-block-pullquote" style="border-color:#cd2653"><blockquote class="has-text-color has-secondary-color"><p>pullquote</p></blockquote></figure>
+<!-- wp:pullquote {"textColor":"secondary"} -->
+<figure class="wp-block-pullquote has-secondary-color has-text-color"><blockquote><p>pullquote</p></blockquote></figure>
```

Comment:
⚠️ Seems like we're losing the `customMainColor` attribute and style here?

Reply:
Hmm, I'm not sure why it's stripping the inline style. I think I need to relook at the v4 migration function. Thanks for spotting that one!

Reply:
Actually it was the v3 migration function. I'm not 100% certain if what I've done is kosher, but it seems that we need to add the block supports `style` attribute to the migrated attributes so that the inline style is added.

I guess the `customMainColor` attribute is stripped because no longer a valid attribute 🤞

Reply:
I might be a bit late to the conversation and missing something so feel free to ignore this if it doesn't apply or help.

When a block's attributes have changed and you need to add a deprecation with a `migrate` function. All the deprecations need to handle migrating the attributes correctly to the block's latest state or attribute definition, if that makes sense.

So if we have a deprecated attribute in a deprecation's attribute definitions, that would need migrating to the latest version of the block's attributes via the deprecation's `migrate` function.

In other words, each deprecation is responsible for bring that version of the block up all the way up to the latest.

Reply:
> When a block's attributes have changed and you need to add a deprecation with a migrate function. All the deprecations need to handle migrating the attributes correctly to the block's latest state or attribute definition, if that makes sense.

I had suppressed an inkling this might be the case. :) Thanks @aaronrobertshaw, I'll have a closer look.